src/lib.rs

```diff
@@ -1199,41 +1200,69 @@ pub trait InPlaceInit<T>: Sized {
     }
 }
 
-#[cfg(feature = "alloc")]
+#[cfg(HAVE_ALLOCATION)]
 impl<T> InPlaceInit<T> for Box<T> {
     #[inline]
     fn try_pin_init<E>(init: impl PinInit<T, E>) -> Result<Pin<Self>, E>
     where
         E: From<AllocError>,
     {
-        Box::try_new_uninit()?.write_pin_init(init)
+        #[cfg(feature = "alloc")]
+        {
+            Box::try_new_uninit()?.write_pin_init(init)
+        }
+        #[cfg(not(feature = "alloc"))]
+        {
+            Box::new_uninit().write_pin_init(init)
+        }
```

Comment:
can you add a `if_cfg!` macro for this and the other instances below?
```rust
macro_rules! if_cfg {
    (($($cond:tt)*) {$($then:tt)*} else {$($otherwise:tt)*}) => {
        #[cfg($($cond)*)]
        { $($then)* }
        #[cfg(not($($cond)*))]
        { $($otherwise)* }
    }
}
```

Reply:
It didn't really work but I extracted the two #[cfg(...)] alternatives into two implementations of the same macro `try_new_init!`